src/app/design/styled-button/styled-button.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input, OnInit} from '@angular/core';
+@Component({
+  selector: 'app-styled-button',
+  templateUrl: './styled-button.component.html',
+  styleUrls: ['./styled-button.component.scss']
+})
+export class StyledButtonComponent implements OnInit {
+
+  @Input() disabled = false;
+  @Input() styleClass: string;
```

Comment:
```suggestion
  @Input() styleClass: 'primary' | 'success' | 'warning' | 'danger' | 'info' = 'primary';
```
Also, name it some the variable something meaningful e.g. `type`.

Maybe use some namingfull names for the types? @MaxiHuHe04